File: frontEnd/Systetica/lib/model/CidadeDTO.dart
import 'package:systetica/model/EstadoDTO.dart';

class CidadeDTO {
  CidadeDTO({
    required this.id,
    required this.nome,
    this.estado,
  });

  late final int id;
  late final String nome;
  EstadoDTO? estado;

  CidadeDTO.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    estado = EstadoDTO.fromJson(json['estado']);
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id;
    _data['nome'] = nome;
    _data['estado'] = estado?.toJson();
    return _data;
  }
}

File: backEnd/Systetica_Api/src/main/java/com/frfs/systetica/dto/CidadeDTO.java
package com.frfs.systetica.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CidadeDTO implements Serializable {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("nome")
    private String nome;

    @JsonProperty("estado")
    private EstadoDTO estado;
}